docs/coding-guidelines.md

```diff
@@ -0,0 +1,69 @@
+# Coding Guidelines
+
+This living document serves to prescribe coding guidelines specific to the WooCommerce Blocks project. Base coding guidelines follow the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [Gutenberg coding standards](https://developer.wordpress.org/block-editor/contributors/develop/coding-guidelines/).
+
+The following sections outline additional patterns and conventions used in the Blocks project.
+
+## CSS
+
+### CSS Class Names
+
+To avoid class name collisions, class names must adhere to the following guidelines, based on the [BEM methodology](https://en.bem.info/methodology/) and [Gutenberg coding standards](https://developer.wordpress.org/block-editor/contributors/develop/coding-guidelines/).
+
+#### Prefixing
+
+As a WordPress plugin, Blocks has to play nicely with other plugins and themes, and WordPress itself. To minimize conflict potential all classes should be prefixed with `.wc-block-`.
+
+#### Naming
+
+As well as using a `wc-block-` prefix, all class names assigned to an element must be prefixed with the name of the package, followed by a dash and the name of the directory in which the component resides. Any descendent of the component's root element must append a dash-delimited descriptor, separated from the base by two consecutive underscores \_\_.
+
+    Root element: wc-block-package-directory
```

Comment:
This seems to render as code in the rich view:

<img src="https://user-images.githubusercontent.com/3616980/73250901-20e6b580-41b8-11ea-93a6-a1de55ae7942.png" width="161" />

Was that in purpose?

Reply:
I'm rewriting this part, but yes I'm trying to make examples stand out from the descriptions.